### PR TITLE
Add net.codelogistics.webapps

### PR DESCRIPTION
**Pull request by eyekay-gh**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: net.codelogistics.webapps.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+    "app-id" : "net.codelogistics.webapps",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "44",
```
Comment (hfiguiere) [inline net.codelogistics.webapps.json:4]: '45' is the current.

File: net.codelogistics.webapps.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "sdk" : "org.gnome.Sdk",
+    "command" : "webapps",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--filesystem=xdg-data"
```
Comment (hfiguiere) [inline net.codelogistics.webapps.json:13]: why do you need this?

**Comment by hfiguiere** [comment]
bot, build net.codelogistics.webapps

File: net.codelogistics.webapps.json (hunk 2 above)
Comment (hfiguiere) [reply net.codelogistics.webapps.json:13]: it's not allowed by default.

Comment (hfiguiere) [reply net.codelogistics.webapps.json:13]: You should use this instead https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.DynamicLauncher.html

Comment (eyekay-gh) [reply net.codelogistics.webapps.json]: Thank you so much, the program now uses portals, so I have removed that line from the manifest.

**Comment by bbhtt** [comment]
bot, build net.codelogistics.webapps

File: net.codelogistics.webapps.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
+{
+    "app-id" : "net.codelogistics.webapps",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "45",
```
Comment (bbhtt) [inline net.codelogistics.webapps.json:4]: I'd suggest updating to 46.

**Comment by bbhtt** [comment]
This seems to redirect to a regular Firefox instance when creating the webapp. I'm not sure if that is intended?

How is that a webapp?

**Comment by eyekay-gh** [comment]
> This seems to redirect to a regular Firefox instance when creating the webapp. I'm not sure if that is intended?
> 
> How is that a webapp?

That should not be possible, the app does not call the system web browser or ask to open a URL at any point.

I just installed the app as a Flatpak on a fresh install, and when I create a new webapp, the Flatpak/ freedesktop portal to create a new desktop launcher appears (as expected) and the desktop file available through the menu calls `flatpak run --command=webapps 'net.codelogistics.webapps'`.

File: net.codelogistics.webapps.json (hunk 3 above)
Comment (eyekay-gh) [reply net.codelogistics.webapps.json]: Done, thank you

**Comment by bbhtt** [comment]
bot, build net.codelogistics.webapps

**Comment by bbhtt** [comment]
Seems like it works but if I do https://mail.google.com it opens a youtube page in the default browser along with the webapp. I assume that's not coming from the app?


https://github.com/flathub/flathub/assets/62639087/6af0757a-ab5b-4a92-8c5d-935dd3ec6b90

**Comment by eyekay-gh** [comment]
> Seems like it works but if I do https://mail.google.com it opens a youtube page in the default browser along with the webapp. I assume that's not coming from the app?
> Screencast.mp4

The issue here is that my app only allows webpages from the original domain name associated with the webapp to be opened inside it. Google apparently loads the youtube.com domain at that point of the login process, which gets opened in the default browser as it does not match the original domain (google.com). 

I tried this as well, you can safely close that tab. It does not affect the sign-in process.

**Pull request closed without merging**